core/operation/src/main/java/uk/gov/gchq/gaffer/named/operation/ParameterDetail.java

```diff
@@ -37,8 +37,9 @@
     private Object defaultValue;
     private Class valueClass;
     private boolean required;
+    private String options;
```

Comment:
Options should be a list or set to work with the dynamic input component in the UI.

Reply:
I've made the requested change, and changed the dedicated unit test and shallowClone test in `AddNamedOperationTest` accordingly. However, in the same file, I can't get the `shouldJsonSerialiseAndDeserialise` test to work any more, due to a trailing whitespace that appears after the close of the LinkedHashMap. Currently the test is as follows:

`
@Override
    public void shouldJsonSerialiseAndDeserialise() {
        Map<String, ParameterDetail> parameters = new HashMap<>();
        List<String> optionsList = Arrays.asList("option1", "option2", "option3");

        parameters.put("testOption", new ParameterDetail("Description", String.class,false, "On", optionsList));

        final AddNamedOperation obj = new AddNamedOperation.Builder()
                .operationChain(OPERATION_CHAIN)
                .description("Test Named Operation")
                .name("Test")
                .overwrite()
                .readAccessRoles(USER)
                .writeAccessRoles(USER)
                .parameters(parameters)
                .score(0)
                .build();

        // When
        final byte[] json = toJson(obj);
        final AddNamedOperation deserialisedObj = fromJson(json);

        final String expected = String.format("{%n" +
                " \"class\" : \"uk.gov.gchq.gaffer.named.operation.AddNamedOperation\",%n" +
                " \"operationName\": \"Test\",%n" +
                " \"description\": \"Test Named Operation\",%n" +
                " \"score\": 0,%n" +
                " \"operationChain\": {" +
                " \"operations\": [{\"class\": \"uk.gov.gchq.gaffer.operation.impl.get.GetAdjacentIds\", \"input\": [{\"class\": \"uk.gov.gchq.gaffer.operation.data.EntitySeed\", \"vertex\" : \"seed\"}]}]},%n" +
                " \"overwriteFlag\" : true,%n" +
                " \"parameters\" : {\"testOption\": {\"description\" :\"Description\", \"defaultValue\": \"On\", \"valueClass\": \"java.lang.String\", \"required\": false, \"options\": \"[option1, option2, option3]\"}},%n" +
                " \"readAccessRoles\" : [ \"User\" ],%n" +
                " \"writeAccessRoles\" : [ \"User\" ]%n" +
                "}");

        // Then
        JsonAssert.assertEquals(expected, new String(json));
        assertNotNull(deserialisedObj);
`

Reply:
Fixed the test issue I was having, needed to switch around which assertEquals I was using in order to realise I was missing some \" markers in the options that weren't being highlighted by JsonAssert.assertEquals